src_cpp/union_find.hpp

```diff
@@ -523,15 +596,38 @@ class UfDecoder{
         std::vector<uint8_t> decoding;
         int bit_count;
         int check_count;
+        tsl::robin_set<int> planar_code_boundary_bits;
+        bool is_planar_code;
         UfDecoder(ldpc::bp::BpSparse& parity_check_matrix): pcm(parity_check_matrix){
             this->bit_count = pcm.n;
             this->check_count = pcm.m;
             this->decoding.resize(this->bit_count);
             this->weighted = false;
+
+            this->is_planar_code = true;
+            for(int i = 0; i<this->pcm.n; i++){
+                int col_weight = this->pcm.get_col_degree(i);
+                if(col_weight > 2){
+                    this->is_planar_code = false;
+                }
+                else if(col_weight == 1){
+                    this->planar_code_boundary_bits.insert(i);
+                }
+                else if(col_weight == 0){
+                    throw(std::runtime_error("Invalid parity check matrix. Column weight is zero."));
+                }
+            }
+
+
         }
 
         std::vector<uint8_t>& peel_decode(const std::vector<uint8_t>& syndrome, const std::vector<double>& bit_weights = NULL_DOUBLE_VECTOR, int bits_per_step = 1){
 
+            if(!this->is_planar_code){
+                throw(std::runtime_error("Peel decoder only works for planar codes. Use the matrix_decode method for more general codes."));
```

Comment:
Why does it not work for non-planar codes? It should work for the Toric code.

Reply:
What's the general name for a code that produces pair-like or pair to boundary excitations.

Reply:
any code with max bit_degree 2 should do that. Planar to me means non-periodic.